File: app/src/main/java/org/thoughtcrime/securesms/deeplinks/DeepLinkEntryActivity.java
package org.thoughtcrime.securesms.deeplinks;

import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;

import org.thoughtcrime.securesms.MainActivity;
import org.thoughtcrime.securesms.PassphraseRequiredActivity;

public class DeepLinkEntryActivity extends PassphraseRequiredActivity {
  @Override
  protected void onCreate(Bundle savedInstanceState, boolean ready) {
    Intent intent = MainActivity.clearTop(this);
    Uri    data   = getIntent().getData();
    intent.setData(data);
    startActivity(intent);
  }
}

File: app/src/main/java/org/thoughtcrime/securesms/deeplinks/DeepLinkEntryActivity.kt
package org.thoughtcrime.securesms.deeplinks

import android.os.Bundle
import org.thoughtcrime.securesms.MainActivity
import org.thoughtcrime.securesms.PassphraseRequiredActivity

class DeepLinkEntryActivity : PassphraseRequiredActivity() {
    override fun onCreate(savedInstanceState: Bundle, ready: Boolean) {
        val intent = MainActivity.clearTop(this)
        val data = getIntent().data
        intent.data = data
        startActivity(intent)
    }
}